textinput/textinput.go

```diff
@@ -899,3 +919,56 @@ func (m Model) validate(v []rune) error {
 	}
 	return nil
 }
+
+func (m *Model) SaveHistory() {
+	if len(m.value) == 0 {
+		return
+	}
+
+	if m.EchoMode != EchoNormal {
+		return
+	}
+
+	// XXX: It seems that doing `append(m.history, m.value)` copies a reference
+	// to the value, which leads to all history items being similar.
+	// This is quite surprising given this other test:
+	// https://go.dev/play/p/MZGpcUzSVch
+	newRecord := make([]rune, len(m.value))
+	for i, r := range m.value {
+		newRecord[i] = r
+	}
+
+	m.history = append(m.history, newRecord)
+	m.historyIndex = len(m.history) - 1
+}
+
+func (m Model) History() []string {
+	// TODO: rename getSuggestions to getStrings
+	return m.getSuggestions(m.history)
```

Comment:
The `m.getSuggestions` really just transforms a `[][]rune` into a `[]string`, so the name could be updated to reflect this.